spring-boot/src/main/java/org/springframework/boot/liquibase/LiquibaseServiceLocatorApplicationListener.java

```diff
@@ -51,12 +52,14 @@ public void onApplicationEvent(ApplicationStartingEvent event) {
 	private static class LiquibasePresent {
 
 		public void replaceServiceLocator() {
-			CustomResolverServiceLocator customResolverServiceLocator = new CustomResolverServiceLocator(
-					new SpringPackageScanClassResolver(logger));
-			customResolverServiceLocator.addPackageToScan(
-					CommonsLoggingLiquibaseLogger.class.getPackage().getName());
-			ServiceLocator.setInstance(customResolverServiceLocator);
-			liquibase.logging.LogFactory.reset();
+			if (ClassUtils.isPresent("liquibase.servicelocator.CustomResolverServiceLocator", null)) {
```

Comment:
Can we please have a test that exercises this behaviour. You can use the `FilteredClassLoader` for that.

Reply:
I think the new check is in the wrong place. I’d expect it to replace the existing check above.

Reply:
Also, given that we use Liquibase 3.5.x in 1.5, it would be worth looking to see if there’s a class that’s new in 3.5 that we can check for. We should review the auto-configuration conditions too.